app-services/secrets/README.md

```diff
@@ -1,36 +1,114 @@
+
+   ```
+   make docker_secrets_example
+   ```
+
+2. Add the environment variable `ADD_SECRETSTORE_TOKENS` to the vault-worker service in your EdgeX docker compose file. It takes a list of service keys that it generates Vault tokens for. This is used as the base path for the service's secrets in Vault.  `secrets-example` is the application service that we will be running.
+
+   ```yml
+   environment:
+         - "SECRETSTORE_SETUP_DONE_FLAG=/tmp/edgex/secrets/edgex-consul/.secretstore-setup-done"
+         - "ADD_SECRETSTORE_TOKENS=secrets-example"
+       
+   ```
+
+3. The default configuration is for running natively and we use environment overrides in the compose files for use in docker.
+
+4. Add the secrets example application service to your EdgeX docker compose file. This includes the environment overrides for the SecretStore and SecretStoreExclusive configurations:
+
+   ```yml
+     app-service-secrets-example:
+       image: edgexfoundry/docker-secrets-example:dev
+       container_name: app-service-secrets-example
+       hostname: app-service-secrets-example
+       ports:
+         - "48095:48095"
+       networks:
+         - edgex-network
+       environment:
+         <<: *common-variables
+         SecretStore_Protocol: https
+         SecretStore_TokenFile: /tmp/edgex/secrets/edgex-application-service/secrets-token.json
+         SecretStoreExclusive_Host: edgex-vault
+         SecretStoreExclusive_Protocol: https
+         SecretStoreExclusive_RootCaCertPath: /tmp/edgex/secrets/ca/ca.pem
+         SecretStoreExclusive_ServerName: edgex-vault
+         SecretStoreExclusive_TokenFile: /tmp/edgex/secrets/secrets-example/secrets-token.json
+       volumes:
+         - /tmp/edgex/secrets/ca:/tmp/edgex/secrets/ca:ro,z
+         - /tmp/edgex/secrets/edgex-application-service:/tmp/edgex/secrets/edgex-application-service:ro,z 
+         - /tmp/edgex/secrets/secrets-example:/tmp/edgex/secrets/secrets-example:ro,z 
+       depends_on:
+         - vault-worker
+   ```
+
+   4. Run EdgeX security services in docker:
+
+      The secrets example in docker is dependent on the EdgeX security services, at a minimum. 
+
+      1. Run the secret store services in EdgeX using:
+
+         `docker-compose -f <docker-compose file name> up -d vault-worker`
+
+         Verify that the following services are running:
+
+         ```
+         edgex-core-consul
+         edgex-secrets-setup
+         edgex-vault        
+         edgex-vault-worker
+         ```
+         
+2. Wait for about a minute to ensure tokens are created for the services that were specified in the docker compose section of vault-worker service for `ADD_SECRETSTORE_TOKENS` environment variable. Verify that the token `secrets-example` exist under `/tmp/edgex/secrets`.
+      
+5. Run the secrets example service:
+   
+   ```
+      docker-compose -f <docker-compose file name> up -d app-service-secrets-example
```

Comment:
This will not work without Core Data running. App service pings Core Data for Version and will fail if Core Data not running.

```
error="Get http://edgex-core-data:48080/api/version: dial tcp: lookup edgex-core-data on 127.0.0.11:53: no such host" msg="Unable to get version of Core Services after retries"
```

Adding ` - data` to the depends_on will resolve this.

Reply:
the App Service doesn't fail if Core Data isn't running. It just keeps logging the error message 'unable to get service endpoint for edgex-core-data: no matching service endpoint found'

Reply:
i'm using the nexus docker image - master tag

Reply:
i can still add `-data` to get rid of the error though

Reply:
I tested it with mine using ASC, it will fail on the version check.